haskell/src/Stablecoin/Client/Contract.hs

```diff
@@ -2,21 +2,37 @@
 -- SPDX-License-Identifier: MIT
 
 module Stablecoin.Client.Contract
-  ( mkInitialStorage
+  ( ContractMetadataOptions(..)
   , InitialStorageData(..)
+  , mkInitialStorage
   ) where
 
-import Michelson.Text (MText)
 import Morley.Client (AddressOrAlias)
-import Tezos.Address (Address)
 
-import Lorentz.Contracts.Stablecoin (Expiry, Roles(..), Storage, Storage'(..), metadataMap)
+import Lorentz as L
+import Lorentz.Contracts.Spec.TZIP16Interface (MetadataMap)
+import Lorentz.Contracts.Stablecoin (Expiry, Roles(..), Storage, Storage'(..))
+
+data ContractMetadataOptions = OpCurrentContract | OpRemoteContract | OpRaw Text
```

Comment:
Originally I didn't put it in Parser module because it appeared to cause circular imports. But surprisingly, I was able to solve it here by moviing something around. So have moved it to Parser module now and have added comments.